src/libcore/macros.rs

```diff
@@ -229,14 +229,26 @@ macro_rules! try {
     })
 }
 
-/// Use the `format!` syntax to write data into a buffer.
+/// Calls `write_fmt` function on a writer
 ///
-/// This macro is typically used with a buffer of `&mut `[`Write`][write].
+/// This macro takes an implementor of [`std::fmt::Write`][fmt_write] or
+/// [`std::io::Write`][io_write] trait, a format string, and a list of arguments.
+///
+/// Implementors of the `Write` trait are sometimes called 'writers'.
+///
+/// These arguments will be formatted according to the specified format string and
+/// the resulting string will be passed to the writer.
+///
+/// Return value is either [`Result`][enum_result] or [`io::Result`][type_result] depending on
```

Comment:
Similarly to above, the return value is anything `write_fmt` method returns. It is nice to have return types of common sources of such method mentioned.